internal/util/osio_token.go

```diff
@@ -0,0 +1,135 @@
+package util
+
+import (
+	"crypto/rsa"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/satori/go.uuid"
+)
+
+type User struct {
+	Data Data `json:"data"`
+}
+
+type Data struct {
+	Attributes Attributes `json:"attributes"`
+}
+
+type Attributes struct {
+	Username string `json:"username"`
+	Email    string `json:"email"`
+	FullName string `json:"fullName"`
+}
+
+type Result struct {
+	Tokens []TokenResult `json:"tokens"`
+}
+
+type TokenResult struct {
+	UUID  string `json:"uuid"`
+	Token string `json:"token"`
+}
+
+type Token struct {
+	Data Data `json:"data"`
+}
+
+func CreateOSIOToken(env string, uuid string, key string, session string) (string, error) {
+	privateKey, err := jwt.ParseRSAPrivateKeyFromPEM([]byte(key))
+	if err != nil {
+		return "", errors.New(fmt.Sprintf("Unable to parse private key file: %s", err))
+	}
+
+	user, err := loadUser(env, uuid)
+	if err != nil {
+		return "", errors.New(fmt.Sprintf("Unable to GET user for uuid '%s': %s", uuid, err))
+	}
+
+	token, err := generateToken(env, privateKey, user, uuid, session)
+	if err != nil {
+		return "", errors.New(fmt.Sprintf("Unable to generate token for uuid '%s': %s", uuid, err))
+	}
+
+	return token, nil
+}
+
+func generateToken(env string, key *rsa.PrivateKey, user User, userID string, session string) (string, error) {
```

Comment:
@alexeykazakov would you mind helping me with this token generation function? This function is a slightly modified version of your token-generation script, but I am having problems getting it to work on prod-preview. It spits out a token, but it does not seem to be valid when I then try to use it to get the GitHub token for this user.

Reply:
FYI, the private key used is 'serviceaccount.privatekey'

Reply:
Here is the error I get, when using the osio token as generated by this function:

    {"errors":[{"code":"unauthorized_error","detail":"There is no public key with such ID: 0lL0vXs9YRVqZMowyw8uNLR_yr0iFaozdQk9rzq2OVU","status":"401","title":"Unauthorized error"}]}

Reply:
See my comment regarding kid (Key ID) below.